src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/ExplosionFurnaceAdditives.java

```diff
@@ -0,0 +1,132 @@
+package com.cleanroommc.groovyscript.compat.mods.prodigytech;
+
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.Example;
+import com.cleanroommc.groovyscript.api.documentation.annotations.MethodDescription;
+import com.cleanroommc.groovyscript.api.documentation.annotations.RegistryDescription;
+import com.cleanroommc.groovyscript.helper.ingredient.ItemsIngredient;
+import com.cleanroommc.groovyscript.helper.ingredient.OreDictIngredient;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import lykrast.prodigytech.common.recipe.ExplosionFurnaceManager;
+import net.minecraft.item.ItemStack;
+
+@RegistryDescription(category = RegistryDescription.Category.ENTRIES)
+public class ExplosionFurnaceAdditives extends VirtualizedRegistry<ExplosionFurnaceAdditives.EFAdditiveRecipe> {
+    @Override
+    public void onReload() {
+        removeScripted().forEach(EFAdditiveRecipe::unregister);
+        restoreFromBackup().forEach(EFAdditiveRecipe::register);
+    }
+
+    private void add(EFAdditiveRecipe recipe) {
+        recipe.register();
+        addScripted(recipe);
+    }
+
+    private void remove(EFAdditiveRecipe recipe) {
+        recipe.unregister();
+        addBackup(recipe);
+    }
+
+    @MethodDescription(example = @Example("item('minecraft:cobblestone'), 50"), type = MethodDescription.Type.ADDITION)
+    public void addExplosive(IIngredient explosive, int power) {
+        EFAdditiveRecipe recipe = new EFAdditiveRecipe(true, explosive, power);
+        add(recipe);
+    }
+
+    @MethodDescription(example = @Example("ore('gunpowder')"))
+    public boolean removeExplosive(IIngredient explosive) {
+        for (ItemStack it : explosive.getMatchingStacks()) {
+            ExplosionFurnaceManager.Explosive externalExplosive = ExplosionFurnaceManager.findExplosive(it);
+            if (externalExplosive == null) continue;
+            EFAdditiveRecipe recipe = new EFAdditiveRecipe(true, explosive, externalExplosive.getPower());
+            remove(recipe);
+            return true;
+        }
+        return false;

```

Comment:
wtf???

Reply:
Yeah uh, I think this might be the only sensible way to do it.

Reply:
It isnt. if you apply the change the convo below it will become much better